File: day05/src/test/java/de/havox_design/aoc2015/day05/Day05Test.java
package de.havox_design.aoc2015.day05;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

class Day05Test {

    @SuppressWarnings("java:S2699")
    @Test
    void testMainClass() {
        MainClass.main(new String[0]);
    }

    @ParameterizedTest
    @MethodSource("getDataForPart1")
    void testPart1(String fileName, int expectedNumberOfNiceStrings) {
        Assertions.assertEquals(expectedNumberOfNiceStrings, NiceStrings.solvePart1(fileName));
    }

    private static Stream<Arguments> getDataForPart1() {
        return Stream.of(
                Arguments.of("part1sample1.txt", 1),
                Arguments.of("part1sample2.txt", 1),
                Arguments.of("part1sample3.txt", 0),
                Arguments.of("part1sample4.txt", 0),
                Arguments.of("part1sample5.txt", 0),
                Arguments.of("part1sample6.txt", 2)
        );
    }
}

File: aoc2023/src/test/kotlin/de/havox_design/aoc2023/day05/Day05Test.kt
package de.havox_design.aoc2023.day05

import de.havox_design.aoc.utils.kotlin.helpers.tests.shouldBe
import de.havox_design.aoc2023.MainClass
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class Day05Test {
    @Test
    fun testMainClass() {
        MainClass.main(arrayOf("day05", "testing"))
    }

    @ParameterizedTest
    @MethodSource("getDataForTestSolvePart1")
    fun testSolvePart1(filename: String, expectedResult: Long) =
        IfYouGiveASeedAFertilizer(filename).solvePart1().shouldBe(expectedResult)

    @ParameterizedTest
    @MethodSource("getDataForTestSolvePart2")
    fun testSolvePart2(filename: String, expectedResult: Long) =
        IfYouGiveASeedAFertilizer(filename).solvePart2().shouldBe(expectedResult)

    companion object {
        @JvmStatic
        private fun getDataForTestSolvePart1(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2023/day05/day05Sample.txt", 35L)
            )

        @JvmStatic
        private fun getDataForTestSolvePart2(): Stream<Arguments> =
            Stream.of(
                Arguments.of("de/havox_design/aoc2023/day05/day05Sample.txt", 46L)
            )
    }
}
